### PR TITLE
CMake: Add support for exported symbols from xlc

### PR DESCRIPTION
**Pull request by dnakamura**
Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

File: cmake/modules/platform/toolcfg/xlc.cmake [hunk 1]
```diff
@@ -185,3 +185,19 @@ endif()
 
 set(SPP_CMD ${CMAKE_C_COMPILER})
 set(SPP_FLAGS -E -P)
+
+function(_omr_toolchain_process_exports TARGET_NAME)
```
Comment (fjeremic) [inline cmake/modules/platform/toolcfg/xlc.cmake]: We seem to have this function defined for z/OS slightly above in [1]. Does this new definition make the old one obsolete? If so we should remove the old version.

[1] https://github.com/eclipse/omr/blob/7c8742285db4a1db3b3d55fe6f04cfc80129f20b/cmake/modules/platform/toolcfg/xlc.cmake#L172-L183

Comment (dnakamura) [reply cmake/modules/platform/toolcfg/xlc.cmake]: Yeah, I just noticed that as well. I'm looking now to see if the export format is the same.

Comment (charliegracie) [reply cmake/modules/platform/toolcfg/xlc.cmake]: Any update @dnakamura ?

**Comment by charliegracie** [comment]
@genie-omr build all

File: cmake/modules/platform/toolcfg/xlc.cmake [hunk 2]
```diff
@@ -168,20 +168,38 @@ elseif(OMR_OS_ZOS)
 
 	# Configure the platform dependent library for multithreading
 	set(OMR_PLATFORM_THREAD_LIBRARY "")
+endif()
+
+set(SPP_CMD ${CMAKE_C_COMPILER})
+set(SPP_FLAGS -E -P)
 
+if(OMR_OS_ZOS)
 	function(_omr_toolchain_process_exports TARGET_NAME)
```
Comment (rwy0717) [inline cmake/modules/platform/toolcfg/xlc.cmake]: The body of this function should be indented

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by dnakamura** [comment]
Updated to use our template processing function from #4114

**Comment by youngar** [comment]
@genie-omr build all

File: cmake/modules/platform/toolcfg/xlc.cmake [hunk 3]
```diff
@@ -181,7 +186,19 @@ elseif(OMR_OS_ZOS)
...
+		get_target_property(target_type ${TARGET_NAME} TYPE)
+		if(NOT target_type STREQUAL "SHARED_LIBRARY")
+			return()
+		endif()
 
-set(SPP_CMD ${CMAKE_C_COMPILER})
-set(SPP_FLAGS -E -P)
+		set(exp_file "$<TARGET_PROPERTY:${TARGET_NAME},BINARY_DIR>/${TARGET_NAME}.exp")
+		omr_process_template(
+			"${omr_SOURCE_DIR}/cmake/modules/platform/toolcfg/xlc_exports.exp.in"
```
Comment (rwy0717) [inline cmake/modules/platform/toolcfg/xlc.cmake]:
```suggestion
			"${CMAKE_CURRENT_LIST_DIR}/xlc_exports.exp.in"
```

Comment (dnakamura) [reply cmake/modules/platform/toolcfg/xlc.cmake]: From CMake Docs:
>Therefore the value of the variable inside a macro or function is the directory of the file invoking the bottom-most entry on the call stack, not the directory of the file containing the macro or function definition

Comment (rwy0717) [reply cmake/modules/platform/toolcfg/xlc.cmake]: Good call! I did not know that!

**Pull request merged**